src/core/NodeCache.h

```diff
@@ -36,16 +36,38 @@ class NodeCache
 
   void insertStart(const size_t nodeidx, const long startindex)
   {
-    assert(this->cache.count(nodeidx) == 0 && "start index inserted twice");
-    this->cache.emplace(nodeidx, std::make_pair(startindex, -1L));
+#ifdef ENABLE_PYTHON
+    if (true) {
+      if (this->cache.count(nodeidx) ==
+          0)  // with python it can happen that nodes get dumped several times,
+              // but its understood that the dump will always be identical
```

Comment:
I don't quote understand why. This sounds like a failure to do a cache lookup before inserting into the cache?